pym/portage/dbapi/porttree.py

```diff
@@ -255,6 +255,48 @@ def __init__(self, _unused_param=DeprecationWarning, mysettings=None):
 		self._aux_cache = {}
 		self._broken_ebuilds = set()
 
+		# The purpose of self.better_cache is to perform an initial quick scan of all repositories
+		# using os.listdir, which is less expensive IO-wise. Then it compiles a list of repos in
+		# which particular catpkgs appear. For example, better_cache data may look like this:
+		#
+		# { "sys-apps/portage" : [ repo1, repo2 ] }
+		#
+		# This information can then be used by other functions like aux_get and findname2 to avoid
+		# doing an exhaustive search of repositories for ebuilds. Without this tweak, Portage will
+		# get slower and slower as more overlays are added.
+		#
+		# Also note that it is OK if this cache has some 'false positive' catpkgs in it. We use it
+		# to search for specific catpkgs listed in ebuilds. The likelihood of a false positive catpkg
+		# in our cache causing a problem is extremely low. Thus, the code below is optimized for
+		# speed rather than correctness.
+
+		self.better_cache = {}
+
+		for repo_loc in reversed(self.porttrees):
+			repo = self.repositories.get_repo_for_location(repo_loc)
+			try:
+				for d in os.listdir(repo_loc):
+					if d[0] == "." or d[0] == "-":
+						continue
+					cat_dir = repo_loc + "/" + d
+					if not os.path.isdir(cat_dir):
+						continue
+					try:
+						for p in os.listdir(cat_dir):
+							catpkg_dir = cat_dir + "/" + p
+							if not os.path.isdir(catpkg_dir):
+								continue
+							catpkg = d + "/" + p
+							if not catpkg in self.better_cache:
+								self.better_cache[catpkg] = []
```

Comment:
`self._better_cache = collections.defaultdict(list)` eliminates the above 2 lines.